spec/algorithms/systematic_sampling_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe Split::Algorithms::SystematicSampling do
+  let(:experiment) do
+    Split::Experiment.new(
+      'link_color',
+      :alternatives => ['red', 'blue', 'green'],
+      :algorithm => Split::Algorithms::SystematicSampling,
+      :cohorting_block_magnitude => 2
+    )
+  end
+
+  it "should return an alternative" do
+    expect(Split::Algorithms::SystematicSampling.choose_alternative(experiment).class).to eq(Split::Alternative)
+  end
+
+  context "experiments with a random seed" do
+    it "cohorts the first block of users equally into each alternative" do
+      results = {'red' => 0, 'blue' => 0, 'green' => 0}
+      6.times do
+        results[Split::Algorithms::SystematicSampling.choose_alternative(experiment).name] += 1
+      end
+
+      expect(experiment.cohorting_block_magnitude * experiment.alternatives.length).to eq(6)
```

Comment:
Are you trying to make sure the `choose_alternative` won't change the value of `cohorting_block_magnitude`? I didn't quite get what you want to secure here...